pkg/infra/teams/team3/fight_strat.go

```diff
@@ -160,61 +170,43 @@ func (a *AgentThree) HandleFightProposal(m message.Proposal[decision.FightAction
 }
 
 func (a *AgentThree) thresholdDecision(baseAgent agent.BaseAgent, choice decision.FightAction) (float64, float64, float64, float64) {
-	// view := baseAgent.View()
-	// agentState := baseAgent.AgentState()
 	HPThreshold, StaminaThreshold, AttackThreshold, DefenseThreshold := 0.0, 0.0, 0.0, 0.0
+	// initiate modifers
+	alpha := 0.2
+	beta := 0.1
+	// get my stats
+	myHP, myST, myATT, myDEF := a.getMyStats(baseAgent)
+	// get group stats
+	groupAvHP, groupAvST, groupAvATT, groupAvDEF := a.getGroupAvStats(baseAgent)
+
+	// get differences (group to me)
+	Delta1HP := groupAvHP - float64(myHP)
+	Delta1ST := groupAvST - float64(myST)
+	Delta1ATT := groupAvDEF - float64(myATT)
+	Delta1DEF := groupAvATT - float64(myDEF)
+
+	if len(a.TSN) > 0 {
+		// get TSN average stats
+		TSNavHP, TSNavST, TSNavATT, TSNavDEF := a.getTSNAvStats(baseAgent)
+		// get differences (group to TSN)
+		Delta2HP := groupAvHP - TSNavHP
+		Delta2ST := groupAvST - TSNavST
+		Delta2ATT := groupAvATT - TSNavATT
+		Delta2DEF := groupAvDEF - TSNavDEF
+
+		HPThreshold = myHP + alpha*Delta1HP + beta*Delta2HP
+		StaminaThreshold = myST + alpha*Delta1ST + beta*Delta2ST
+		AttackThreshold = myATT + alpha*Delta1ATT + beta*Delta2ATT
+		DefenseThreshold = myDEF + alpha*Delta1DEF + beta*Delta2DEF
+
+		return HPThreshold, StaminaThreshold, AttackThreshold, DefenseThreshold
```

Comment:
worth wrapping this in a `threshold struct` for clarity (multivariate returns are a bit gross)

Reply:
yep ok